functions.php

```diff
@@ -66,31 +66,38 @@ function listPlantTypes(array $plant_types): string {
     return $type_echo;
 }
 
-function checkforUniqueAddEntry($db, string $science_name): bool {
-    $nonunique_entry_query = $db->prepare("SELECT `id` FROM `plants` WHERE `science_name` = ?;");
-    $nonunique_entry_query->execute([$science_name]);
+function checkforUniqueAddEntry($db, string $science_name, string $common_name): bool {
+    $nonunique_entry_query = $db->prepare("SELECT `id` FROM `plants` WHERE `science_name` = ? OR `name` = ?;");
+    $nonunique_entry_query->execute([$science_name, $common_name]);
     $nonunique_entry = $nonunique_entry_query->fetch();
-    if ($nonunique_entry !== false) {
-        $nonunique_entry = true;
-    }
+    $nonunique_entry = !$nonunique_entry ? false : true;
 
     return $nonunique_entry;
 }
 
-function checkForWrongPlantType($db, string $type): array {
-    $used_type_query = $db->prepare("SELECT `id` FROM `plant_types` WHERE `type` = upper(?);");
-    $used_type_query->execute([$type]);
+function checkForWrongPlantType($db, string $type): int {
+    $upper_type = strtoupper($type);
+    $used_type_query = $db->prepare("SELECT `id` FROM `plant_types` WHERE `type` = ?;");
+    $used_type_query->execute([$upper_type]);
     $used_type = $used_type_query->fetch();
-    if ($used_type === false) {
-        $used_type = ['not found'];
-    }
+    $type_int = (int) $used_type['id'];
 
-    return $used_type;
+    return $type_int;
 }
 
-function insertDataToDB($db, string $science_name, string $name, array $used_type): string {
-    $add_entry_query = $db->prepare("INSERT INTO `plants` (`science_name`,`name`,`type`) VALUES (?,?,?);");
-    $add_entry_query->execute([$science_name, $name, $used_type['id']]);
+function insertDataToDB($db, array $plantData): string {
+    $add_entry_query = $db->prepare("INSERT INTO `plants` (`science_name`,`name`,`type`) VALUES (:science_name,:common_name,:type_int);");
+    $add_entry_query->execute($plantData);
```

Comment:
execute will return a boolean as to if the insert worked or not, so best to just return this line, then do an if on the result